File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.java
class Solution {
    public int specialArray(int[] nums) {
        Arrays.sort(nums);
        int l = nums.length , curr =0 , j =0;

        //Base Condition
        if(l == 0) return 0;
        if(l == 1) return (nums[0] >=1) ? 1 : -1;
   
        
        for(int i =0 ;i <= l; i++){
            if(i < nums[j]){
                if( (l - j) == i){
                    return i;
                }
            }
            else if(i == nums[j]){
                if((l-j) == i){
                    return i;
                }
                while( j < l && nums[j] <= i)
                    j++;
                if(j >= l)  return -1;
            }
            else    j++;
        }
        return -1; 
    }
}

File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        nums.sort()
        n=len(nums)
        if n==0 and nums[0]==0:
            return 0
        i=0
        c=0
        t=0
       
        while i<n:
            if nums[i]==0:
                i+=1
            else:
                c+=1
              
                i+=1
                
        for ele in nums:
            if ele<c and ele!=0:
                c-=1
        for ele in nums:
            if ele>=c:
                t+=1



        if c!=0 and t==c:
            return c   
        return -1